src/SpotifyRemote.ts

```diff
@@ -253,20 +252,94 @@ export interface SpotifyRemoteApi extends TypedEventEmitter<SpotifyRemoteEvents>
     getCrossfadeState(): Promise<CrossfadeState>;
 }
 
-/**
- * @ignore
- */
-const SpotifyRemote = NativeModules.RNSpotifyRemoteAppRemote as SpotifyRemoteApi;
-RNEvents.register(SpotifyRemote);
-RNEvents.conform(SpotifyRemote);
-
-// Example of Javascript only api method
-SpotifyRemote.setPlaying = (playing: boolean) => {
-    // todo: Will want to likely check the state of playing somewhere?
-    // Perhaps this can be done in native land so that we don't need to
-    // worry about it here
-    return playing ? SpotifyRemote.resume() : SpotifyRemote.pause();
-}
+const nativeModule = NativeModules.RNSpotifyRemoteAppRemote;
+
+const nativeEventEmitter = new NativeEventEmitter(nativeModule);
+const eventListeners = new Set<EmitterSubscription>();
```

Comment:
```suggestion
type SpotifyEventType = keyof SpotifyRemoteEvents;
const eventListeners = new Map<SpotifyEventType,EmitterSubscription[]>();
```

instead, that way the event listeners are bucketed by event.